src/main/java/fr/sncf/osrd/infra/routegraph/RouteGraph.java

```diff
@@ -30,6 +30,19 @@
         return node.getRouteNeighbors(nodeDirection);
     }
 
+    /**
+     * @param route The route for which we want the list of neighbors
+     * @return The neighbors of the route at the start node
+     */
+    public List<Route> getIncomingNeighbors(Route route) {
+        var firstTvdSectionPathIndex = 0;
+        var firstTvdSectionPath = route.tvdSectionsPaths.get(firstTvdSectionPathIndex);
+        var firstTvdSectionPathDir = route.tvdSectionsPathDirections.get(firstTvdSectionPathIndex);
+        var node = getNode(firstTvdSectionPath.getStartNode(firstTvdSectionPathDir));
+        var nodeDirection = firstTvdSectionPath.nodeDirection(firstTvdSectionPathDir, EdgeEndpoint.BEGIN);
+        return node.getIncomingRouteNeighbors(nodeDirection);
+    }
+
```

Comment:
It was used when `PreviousReservedRoute` was given a `Route` parameter. I intentionaly leave it because I was thinking that it is a potentially usefull getter.
I have no problem deleting it of course.